lib/src/arithmetic/multiplier.dart

```diff
@@ -13,6 +13,98 @@ import 'package:meta/meta.dart';
 import 'package:rohd/rohd.dart';
 import 'package:rohd_hcl/rohd_hcl.dart';
 
+/// A general configuration class for specifying parameters that are
+/// for both static or runtime configurations of a component feature.
+class Config {
+  /// The runtime configuration logic that can be used to configure the
+  /// component at runtime
+  final Logic? runtimeConfig;
+
+  /// The static configuration flag that indicates whether the
+  /// feature is statically configured or not.
+  late final bool staticConfig;
+
+  /// The name of the configuration, especially needed for runtime to add as
+  /// a module input.
+  final String name;
+
+  /// Creates a new [Config] instance.
+  Config({required this.name, this.runtimeConfig, bool? staticConfig = false}) {
+    if (runtimeConfig != null && staticConfig != null) {
+      throw RohdHclException(
+          'Only provide either runtimeConfig or staticConfig, not both.');
+    } else if (staticConfig != null) {
+      this.staticConfig = staticConfig;
+    } else {
+      this.staticConfig = false;
+    }
+  }
+
+  /// Factory constructor to create a [Config] instance from a dynamic.
+  factory Config.ofDynamic(dynamic config) {
+    if (config is Config) {
+      return config;
+    } else if (config is bool) {
+      return BooleanConfig(staticConfig: config);
+    } else if (config == null) {
+      return BooleanConfig(staticConfig: null);
+    } else if (config is Logic) {
+      return RuntimeConfig(config, name: config.name);
+    } else {
+      throw RohdHclException(
+          'Unsupported configuration type: ${config.runtimeType}');
+    }
+  }
+
+  /// Return a bool representing the value of the configuration.
+  @visibleForTesting
+  bool get value =>
+      staticConfig ||
+      (runtimeConfig != null && runtimeConfig!.value == LogicValue.one);
+
+  /// Return the internal [Logic] signal that represents the configuration,
+  /// either static or runtime.
+  Logic getLogic(Module module) =>
+      staticConfig ? Const(1) : (tryRuntimeInput(module) ?? Const(0));
```

Comment:
why not `getRuntimeInput` instead of try here?  That way we don't need to call `getRuntimeInput` ahead of time before calling `getLogic`?